### ISSUE TITLE
L046: Only triggers for first Jinja tag in a compound statement with multiple tags

### ISSUE THREAD

**Issue by dmohns** [sqlfluff/sqlfluff#3275]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When linting a SQL with a given statement consists of multiple Jinja tags, only the first Jinja tags is subject to L046.

See an example for BigQuery

```sql
CREATE TABLE `{{ project }}.{{ dataset }}.{{table}}`;
```

--> No Lint error

However, for 

```sql
CREATE TABLE `{{project}}.{{ dataset }}.{{table}}`;
```

```
L:   1 | P:  14 | L046 | Jinja tags should have a single whitespace on either
                       | side.
```

### Expected Behaviour

I would expect an L046 to be raised because `{{table}}` is clearly violating L046.

### Observed Behaviour

No lint error reported.

### How to reproduce

Run `sqlfluff lint --dialect bigquery 1.sql` on the example above

### Dialect

bigquery (but probably not related to bigquery)

### Version

latest `main` from Github

### Configuration

[sqlfluff:templater:jinja:context]
project=project_id
dataset=dataset_id
table=table_name

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
